Implementation of Non-Result Board Methods with Tests

okonekp opened the pull request:
Implemented Board class, ship class pulled into this branch is already reviewed and tested.

AlexLeam commented:
The code looks good. In the tests, doesn't just using assert have the potential to crash?

AlexLeam commented on src/test/java/cs361/battleships/models/BoardTest.java:
```diff
@@ -3,12 +3,56 @@
+        Board board = new Board();
+        List<Ship> ships = new ArrayList<Ship>();
+        ships.add(new Ship("BATTLESHIP"));
+        assert(board.getShips().size() == 0);
```
This probably won't pass seeing as how you're adding a ship then checking if it's empty

okonekp commented:
Changed bounds for the placement checking function to be correct; also removed some print lines in the testing functions that were used for testing the tests.

okonekp commented:
Administrative note: when this is merged, it will resolve #2
